ChaosMod/Effects/db/Misc/MiscCocktail.cpp

```diff
@@ -0,0 +1,49 @@
+//
+// effect by veloscocity
+//
+
+#include "Memory/Gravity.h"
+#include <math.h>
+#include <stdafx.h>
+
+#define PI 3.14159265
+
+static float x, y;
+
+static DWORD lastTick = 0;
+static void OnStart()
+{
+	Ped playerPed = PLAYER_PED_ID();
+	float rot =
+	    GET_ENTITY_HEADING(IS_PED_IN_ANY_VEHICLE(playerPed, false) ? GET_VEHICLE_PED_IS_IN(playerPed, false)
+	                                                               : playerPed); // Shake perpendicular to the player
+	x = sin((360 - rot) * PI / 180) * 1.33;
+	y = -cos((360 - rot) * PI / 180) * 1.33;
```

Comment:
```suggestion
	x = sin((360 - rot) * PI / 180) * 1.33f;
	y = -cos((360 - rot) * PI / 180) * 1.33f;
```
Don't think we need doubles here

Reply:
precomputing pi/180 would save some CPU cycles.

Reply:
PI us defined in multiple other effects, could it be worth adding it to a common header file? Of course PI won't change, though it would still feel ""cleaner"" to have it defined in one place.

Reply:
I believe it's better to use SYSTEM::SIN and SYSTEM::COS natives: they work with angles in degrees, so you won't have to use the value of pi.

Reply:
`math.h` of DirectX has all of that. But I don't see one for PI/180

Value definitions: https://learn.microsoft.com/en-us/cpp/c-runtime-library/math-constants?view=msvc-170#remarks

Reply:
we could (and probably should) just create our own header file that has definitions for all the possible numbers we could need and just include that file in stdafx.h.

Reply:
so what am i supposed to do now